qos/evm/request_validator.go

```diff
@@ -0,0 +1,170 @@
+package evm
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/gateway"
+	qosobservations "github.com/buildwithgrove/path/observation/qos"
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+// maximum length of the error message stored in request validation failure observations and logs.
+const maxErrMessageLen = 1000
+
+// requestValidator handles EVM request validation, generating appropriate error contexts
```

Comment:
Multiple things can happen here:

1. We refactor out `requestValidator` from `qos/evm` to `qos`
- Keep the `requestValidator` as is but need to update the comment to remove `EVM` as we're 100% going to forget to update it

3. We copy-pasta this into other `qos/XXX` packages
- Rename `requestValidator` to `requestValidatorEVM` to guarantee no issues
- Update the comment which copy-pastesr will definitely forget to update